modules/renter/hostdb/scan.go

```diff
@@ -275,19 +296,31 @@ func (hdb *HostDB) managedScanHost(entry modules.HostDBEntry) {
 		copy(pubkey[:], pubKey.Key)
 		return crypto.ReadSignedObject(conn, &settings, maxSettingsLen, pubkey)
 	}()
+	success := false
 	if err != nil {
 		hdb.log.Debugf("Scan of host at %v failed: %v", netAddr, err)
 
 	} else {
 		hdb.log.Debugf("Scan of host at %v succeeded.", netAddr)
 		entry.HostExternalSettings = settings
+		success = true
 	}
 
+	hdb.mu.Lock()
+	defer hdb.mu.Unlock()
 	// Update the host tree to have a new entry, including the new error. Then
 	// delete the entry from the scan map as the scan has been successful.
-	hdb.mu.Lock()
 	hdb.updateEntry(entry, err)
-	hdb.mu.Unlock()
+
+	// Update the initial scan timeout if the scan was successful and if the
+	// latency for this scan was greater than the current timeout. We want to
+	// find the worst case latency.
+	if success {
+		hdb.successfulScans++
+		if latency > hdb.initialScanTimeout {
+			hdb.initialScanTimeout = 2 * latency
```

Comment:
May want to add a comment explaining why the `if` compares them directly, but the update uses `latency * 2`